api/swagger.yaml

```diff
@@ -6302,6 +6315,75 @@ definitions:
     additionalProperties:
       type: "string"
 
+  ImageManifestSummary:
+    x-nullable: false
+    description: |
+      ImageManifestSummary represents a summary of an image manifest.
+    type: "object"
+    required: ["Id", "Descriptor", "Available", "ContentSize", "Kind"]
+    properties:
+      Id:
+        description: |
+          ID is the content-addressable ID of an image and is the same as the
+          digest of the image manifest.
+        type: "string"
+        example: "sha256:95869fbcf224d947ace8d61d0e931d49e31bb7fc67fffbbe9c3198c33aa8e93f"
+      Descriptor:
+        $ref: "#/definitions/OCIDescriptor"
+      Available:
+        description: Indicates whether all the children content is available.
+        type: "boolean"
+        example: true
+      ContentSize:
+        description: |
+          The size of the available distributable (possibly compressed) image content
+          in bytes.
+        type: "integer"
+        format: "int64"
+        example: 3987495
+      Kind:
+        description: |
+          The kind of the manifest.
+        type: "string"
+        example: "image"
+        enum:
+          - "image"
+          - "attestation"
```

Comment:
The description of `Manifests` makes it sound like this field will contain _everything_, but this field makes me wonder if we're maybe still filtering?  For example, this doesn't map to specific `mediaType` values (or `aritfactType` values), so must be implied by the BuildKit-supplied annotations, which means we're doing some amount of parsing/"editorializing" of this data, but hopefully not _filtering_ it?  What do we do with objects we don't recognize here? (WASI images, etc)

Reply:
`Manifests` will contain everything, but not all items may `*Data` field set.
In case where the manifest is not recognizable to us as anything, the `Kind` will be `unknown` and no extra `*Data` field will be set.

(Good catch though - I forgot to document the `unknown` value here 😅)

Reply:
It's probably fine to touch-up in a follow-up, but if you're planning to still document these, I think we can either use a bullet-list, like this one (but that one should have a white-space before the list :joy:) https://github.com/moby/moby/blob/1dd102e9b64e56debec06579fa08210c18f0f37a/api/swagger.yaml#L6484-L6489

Or even a small table, like https://github.com/moby/moby/blob/1dd102e9b64e56debec06579fa08210c18f0f37a/api/swagger.yaml#L4302-L4310